cncserver.js

```diff
@@ -2006,32 +2042,12 @@ function connectSerial(options){
     var portNames = ['None'];
     if (cncserver.gConf.get('debug')) console.log('Full Available Port Data:', ports);
 
-    var botMaker = cncserver.botConf.get('controller').manufacturer.toLowerCase();
-    var botProductId = cncserver.botConf.get('controller').productId.toLowerCase();
     for (var portID in ports){
-      var portMaker = (ports[portID].manufacturer || "").toLowerCase();
-      var portProductId = (ports[portID].productId || "").toLowerCase();
       portNames[portID] = ports[portID].comName;
-
-      // OS specific board detection based on current output from serialport 2.0.5
-      switch (process.platform) {
-        case 'win32':
-          // Match by manufacturer partial only.
-          if (portMaker.indexOf(botMaker) !== -1 && autoDetect) {
-            cncserver.gConf.set('serialPath', portNames[portID]);
-          }
-
-          break;
-        case 'darwin':
-        case 'linux':
-        default:
-          // Match by exact productID and Manufacturer (Mac/*nix).
-          if (portMaker == botMaker && portProductId == botProductId && autoDetect) {
-            cncserver.gConf.set('serialPath', portNames[portID]);
-          }
+      if(autoDetect) {
+        autoDetectPort(ports[portID], cncserver.botConf.get('controller'));
```

Comment:
If we're going to refactor this, we likely should make `autoDetectPort()` more atomic, and have it _return_ a single autodetected port instead of operating on the global, and set the global on this line depending on the output. This will require moving the loop to the function, and should streamline and simplify port detection in this section that's always been a bit of a mess.